### DISCUSSION TITLE
Migration in home assistant from "beta" to stable

### DISCUSSION THREAD

**Discussion by Condla** [music-assistant/support#2786] [Q&A] [answered]
I'm running mass as hass add on and I initially installed the beta version. Now I'd like to rather use the stable branch. As a migration I simply tried installing the stable version, stopping the beta and starting the stable and I was hoping it would pick up all my config and players, but it didn't. 
Can someone help me to cleanly migrate this without having to reconfigure everything again?

**Comment by OzGav** [comment, marked as answer]
Unfortunately no there is no way to migrate the config as they are in separate containers. There is a feature on the backlog to be able to export playlists so they can be kept but the ETA for that is not known. In the interim you could copy all your playlists to a local disk playlist or a streaming provider and then you will be able to pull them back in later.

**Comment by rollyfx** [comment]
@Condla @OzGav , I managed to migrate the configuration from the stable version to the beta version.
I created a backup of the stable and beta module from the HomeAssistant backups tool, and replaced the files in the zip before uploading the modified backup and restoring it.
Not sure if this is a good practice but it allowed me to test the beta version without reconfiguring everything.

**Reply by OzGav** [reply]
Might be advanced for some but thanks!

**Reply by pantherale0** [reply]
Just "migrated" from stable to beta and went without issue.  Take a backup of both as @rollyfx said and download locally, extract the tar.gz that is nested within:
![image](https://github.com/user-attachments/assets/0225f4fd-44f8-4e2b-97f2-f2625de9a477)
Delete data folder from the nested archive in the beta backup, and copy the same data folder over from the stable backup
![image](https://github.com/user-attachments/assets/0f1dda14-fa8c-497b-8920-4bf30b170438)
Compress everything back together and upload to HA and restore like a normal backup.

**Reply by Controli-pl** [reply]
> Might be advanced for some but thanks!

Or, if you have a Debian based installation you can make a copy directly in the terminal.

Just stop both add-ons in HA
Delete the folder "d5369777_music_assistant" (For the new add-on) 
rm -r d5369777_music_assistant

Copy the contents from the old to the new one:
cp -rf /usr/share/hassio/addons/data/d5369777_music_assistant_beta/ /usr/share/hassio/addons/data/d5369777_music_assistant/

Check if everything is in place (playlists, players, integration) uninstall the beta version in HA and that's it ;-)

**Reply by blackscreener** [reply]
> Might be advanced for some but thanks!

For the Docker version:
```
docker run --rm \
  -v /mnt/data/supervisor/apps/data/d5369777_music_assistant_beta:/from:ro \
  -v /mnt/data/supervisor/apps/data/d5369777_music_assistant:/to \
  alpine sh -c 'rm -rf /to/* && cp -a /from/. /to/'
```

**Reply by OzGav** [reply]
People should not do this now as the database is generally incompatible between stable and beta. Swapping from beta to stable is particularly ill advised.
